metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/ProfileBuilderBolt.java

```diff
@@ -243,8 +244,18 @@ private void flush(Tuple tickTuple) {
     measurement.setValue(result);
     emit(measurement, tickTuple);
 
+    // Execute the update with the old state
+    Map<String, String> tickUpdate = profileConfig.getTickUpdate();
+    Map<String, Object> state = executor.getState();
+    if(tickUpdate != null) {
+      tickUpdate.forEach((var, expr) -> executor.assign(var, expr, executor.getState()));
```

Comment:
This is what you had in a prior iteration that does what you are describing.

```
profileConfig.getTickUpdate().forEach((var, expr) -> executor.assign(var, expr, new HashMap<>()));
```

The only change that i suggested previously was to make the `result` available to the `tick` expressions, which I think could be generally useful.

```
profileConfig.getTickUpdate().forEach((var, expr) -> executor.assign(var, expr, Collections.singletonMap("result", result));
```